### PR TITLE
repairing the total balance calculation

### LINKED ISSUE #94: Repair the balance calculation and redesign the presentation
# Description
The global TLOS balance of the account is not well calculated because some of the many specific values that compose the total balance are not taken into account.
The following is an exhaustive list of such values but only used one are checked:
- [x] Liquid balance (native)
- [x] Liquid balance (EVM) 
- [x] Rex staked
- [ ] Rex Profit
- [ ] Rex liquid deposit
- [ ] Staked for CPU
- [ ] Staked for NET
- [ ] Refunding from Staking

The different values, not only are not taken into account in the overall balance calculation but also it is confusing in the way they are displayed. In particular, the first balance in the list includes both "Rex Staked" (only) and "Liquid TLOS" (see the first screenshot), but if you press the button in the middle to actually send native TLOS to EVM, only shows the liquid balance (see the second screenshot).

### Screenshots

image

image

## Acceptance Criteria
- Focus on **just** fixing the balance without adding any new features
- Repair the calculation of the global TLOS balance taking into account all the above-mentioned values.
- Replace the current value in the first row of the coins for the liquid TLOS balance ONLY.
- Below the overall amount in dollars, show the total TLOS balance, the resources summary, the rex total balance, and the telos price.

### PR DESCRIPTION
**Pull request by Viterbo**
# Fixes #94 

## Description
The calculation was fixed as described in issue #94. The display part was not implemented due to a previous discussion where we agreed to skip that part because it will be refactored anyway in the near future.

## Test scenarios
Let's call `myaccount` any account you control with the following balances:
- has some TLOS staking for NET and CPU
- some refunding TLOS (when you unstake)
- some TLOS staked in REX with unclaimed profits
- some liquid deposits (funds) in REX,
- some liquid in EVM,
- and finally some liquid TLOS. 

### test scenarios 1
- Log in using  `myaccount` in web-wallet
- explore the same account using OBE.
- The total balance should only differ in Liquid EVM amount

### test scenarios 2
- Log in using  `myaccount` in web-wallet
- Move any TLOS from any of the balances without losing ownership (do not transfer)
- Refresh the website
- The total balance should be the same as before the balance movement.

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented on my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages

### CODE REVIEW

**Review by Kersten-TCD — APPROVED** [review]
The balances seem to show correctly now. I have verified with EOS-Authority which is a match, and OBE (2 TLOS difference on my tested account though which equals staked resources), needs to be address separately for OBE @karynemayer @donnyquixotic 

For the REX balance there is an issue with an edge case, where an account holds REX and REX Savings as the full amount of both is shown (no differentiation inside the balance), which can not be withdrawn with the web wallet and thus leads to an error when trying to remove all. This should be fixed with full code refactoring for Telos Station. @Viterbo

There is still an issue with "strange tokens" showing up though, both on Safari and Chrome, which should be fixed in the next sprint.

Bildschirm­foto 2023-01-05 um 11 19 13

File: src/store/account/actions.js [hunk 1]
```diff
@@ -114,6 +116,16 @@ export const logout = async function({ commit }) {
   }
 };
 
+
+export const getAccountData = async function({ commit, dispatch }) {
+  if (!this.state.account.accountName) {
+    return;
```
Comment (EJSG) [inline src/store/account/actions.js:122]: in this case i think `commit("setData", undefined);` should be called

**Review by Kersten-TCD — APPROVED** [review]
works as expected for Balance part and change of balances.

**Review by donnyquixotic — APPROVED** [review]

**Review by donnyquixotic — APPROVED** [review]

File: src/store/account/actions.js (hunk 1 above)
Comment (Viterbo) [reply src/store/account/actions.js:123]: done here [723fc65](https://github.com/telosnetwork/telos-wallet/pull/110/commits/723fc65b8e4935f053f3be2dbc2932e05db1aa00)

**Comment by Viterbo** [comment]
> The balances seem to show correctly now. I have verified with EOS-Authority which is a match, and OBE (2 TLOS difference on my tested account though which equals staked resources), needs to be address separately for OBE @karynemayer @donnyquixotic
> 
> For the REX balance there is an issue with an edge case, where an account holds REX and REX Savings as the full amount of both is shown (no differentiation inside the balance), which can not be withdrawn with the web wallet and thus leads to an error when trying to remove all. This should be fixed with full code refactoring for Telos Station. @Viterbo
> 
> There is still an issue with "strange tokens" showing up though, both on Safari and Chrome, which should be fixed in the next sprint.
> 
> Bildschirm­foto 2023-01-05 um 11 19 13

@Kersten-CR,
What you say about REX and saving is correct. The total amount of REX displayed includes everything:
  - Matured (ready to withdraw)
  - Deposits (ready to withdraw)
  - Profits
  - Maturing (not ready)
  - Savings (not ready)

In the near future spring, we can address the refactoring task which will be a definitive and clean solution.

Also, I created this issue (https://github.com/telosnetwork/telos-wallet/issues/126) to address the custom token list problems.

**Pull request merged**